test/gst-vaapi/vpp/deinterlace.py

```diff
@@ -6,37 +6,42 @@
 
 from ....lib import *
 from ..util import *
+from .vpp import VppTest
 
 spec = load_test_spec("vpp", "deinterlace")
+spec_r2r = load_test_spec("vpp", "deinterlace", "r2r")
 
-@slash.requires(have_gst)
-@slash.requires(*have_gst_element("vaapi"))
-@slash.requires(*have_gst_element("vaapipostproc"))
-@slash.requires(*have_gst_element("checksumsink2"))
-@slash.parametrize(*gen_vpp_deinterlace_parameters(spec, ["bob", "weave", "motion-adaptive", "motion-compensated"]))
-@platform_tags(VPP_PLATFORMS)
-def test_default(case, method):
-  params = spec[case].copy()
-  params.update(
-    method = map_deinterlace_method(method),
-    mformat = mapformat(params["format"]),
-    tff = params.get("tff", 1))
+class Deinterlace(VppTest):
+  def before(self):
+    vars(self).update(
+      vpp_element = "deinterlace"
+    )
+    super(Deinterlace, self).before()
 
-  if params["method"] is None:
-    slash.skip_test("{} method not supported".format(method))
+  def init(self, tspec, case, method):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      method = map_deinterlace_method(method),
+      mformat = mapformat(self.format),
+      tff = vars(self).get("tff", 1))
+    if self.method is None:
+      slash.skip_test("{} method not supported".format(method))
 
-  params["decoded"] = get_media()._test_artifact(
-    "{}_deinterlace_{method}_{format}_{width}x{height}"
-    ".yuv".format(case, **params))
+  @slash.parametrize(*gen_vpp_deinterlace_parameters(spec, ["bob", "weave", "motion-adaptive", "motion-compensated"]))
+  @platform_tags(VPP_PLATFORMS)
+  def test_default(self, case, method):
+    self.before()
+    self.init(spec, case, method)
+    self.vpp(case)
 
-  call(
-    "gst-launch-1.0 -vf filesrc location={source} num-buffers={frames}"
-    " ! rawvideoparse format={mformat} width={width} height={height}"
-    " interlaced=true top-field-first={tff}"
-    " ! vaapipostproc format={mformat} width={width} height={height}"
-    " deinterlace-mode=1 deinterlace-method={method} ! checksumsink2"
-    " file-checksum=false frame-checksum=false plane-checksum=false"
-    " dump-output=true dump-location={decoded}".format(**params))
+  @slash.parametrize(*gen_vpp_deinterlace_parameters(spec_r2r, ["bob", "weave", "motion-adaptive", "motion-compensated"]))
+  @platform_tags(VPP_PLATFORMS)
+  def test_r2r(self, case, method):
+    self.before()
+    self.init(spec_r2r, case, method)
+    vars(self).setdefault("r2r", 5)
+    self.vpp(case)
 
-  params.setdefault("metric", dict(type = "md5"))
-  check_metric(**params)
+  def check_metrics(self):
+    vars(self).setdefault("metric", dict(type = "md5"))
```

Comment:
This should be moved to `before` method. See test/gst-vaapi/decode/avc.py for syntax example.